File: test/test_files/java/com/example/utils/Logger.java
package com.example.utils;

/**
 * Simple logger class for demonstration.
 */
public class Logger {
    private String name;
    private boolean enabled;
    
    public Logger(String name) {
        this.name = name;
        this.enabled = true;
    }
    
    public void log(String message) {
        if (enabled) {
            System.out.println("[" + name + "] " + message);
        }
    }
    
    public void disable() {
        this.enabled = false;
    }
}

File: test/test_files/cpp/src/utils/logger.cpp
#include "../../include/utils/logger.h"
#include <iostream>

namespace utils {

Logger::Logger(const std::string& name) : name(name), enabled(true) {}

void Logger::log(const std::string& message) {
    if (enabled) {
        std::cout << "[" << name << "] " << message << std::endl;
    }
}

void Logger::disable() {
    enabled = false;
}

} // namespace utils
